src/mscorlib/shared/System/Number.Formatting.cs

```diff
@@ -109,7 +108,8 @@ public static bool TryFormatInt32(int value, string format, NumberFormatInfo inf
             {
                 NumberBuffer number = default;
                 Int32ToNumber(value, ref number);
-                var sb = new ValueStringBuilder(s_numberToStringScratch);
+                Span<char> stackBuffer = stackalloc char[CharStackBufferSize];
+                var sb = new ValueStringBuilder(stackBuffer);
```

Comment:
@VSadov, why is the compiler complaining about the usage of sb? What's unsafe here?